lib/needle.js

```diff
@@ -432,16 +440,17 @@ Needle.prototype.send_request = function(count, method, uri, config, post_data,
           delete config.headers['content-length']; // in case the original was a multipart POST request.
         }
 
-        if (config.follow_set_cookies && resp.cookies)
-          config.headers['cookie'] = cookies.write(resp.cookies);
+        if (config.follow_set_cookies && resp.cookies) {
+          config.headers['Cookie'] = cookies.write(Object.assign(resp.cookies, cookies.read(headers['set-cookie'])));
```

Comment:
To me this could all be consolidated on line 422. If the user wants to `follow_set_cookies`, then `parse_cookies` has to be true anyways.